app/assets/javascripts/spree/frontend/affirm_checkout.js

```diff
@@ -0,0 +1,74 @@
+(function(){
+  /* only include this setup once */
+  if (!window.AffirmPaymentMethods) {
+
+    /*****************************************************\
+        set the shared checkout data
+    \*****************************************************/
+    affirm.checkout($("#affirm_checkout_payload").data("affirm"));
+
+
+    /* wait for the DOM to be ready */
+    affirm.ui.ready(function(){
+      $(function() {
+
+        /*****************************************************\
+            setup loading and cancel events for the form
+        \*****************************************************/
+        affirm.checkout.on("cancel", function(){
+          $("#checkout_form_payment input.disabled")
```

Comment:
If someone wants to use this in a different way, they can't. This javascript file has to exactly confirm to how the frontend intends to use it in order to be useful. Instead of tying this directly to how spree frontend does this, we can create a javascript object that people can "listen" to, so they can define their own interactions as needed.

We can then use that object in this file for documentation purposes and as an example as how someone might use it in their application.

Users using solidus affirm should not be touching the raw affirm library themselves.